src/map.cpp

```diff
@@ -5695,17 +5695,24 @@ const visibility_variables &map::get_visibility_variables_cache() const {
 
 lit_level map::apparent_light_at( const tripoint &p, const visibility_variables &cache ) const {
     const int dist = rl_dist( g->u.pos(), p );
+    
+    // Being outside of the world bubble overrides Clairvoyance.
+    if(p.x < 0 || p.y < 0) {
```

Comment:
We have `map::inbounds` for this, which works with `tripoint` directly, checks the `z` component as well and checks for the upper bound as well.

Reply:
It's a bad idea to have this check here, though.
`apparent_light_at` is used a lot and is one of the most expensive functions, according to profiler.